refactor: remove redundant task-algo checks (#21)
Substra/orchestrator: docs/assets/computetask.md

## Compute task outputs

Upon successful execution, compute task should generate *outputs* which correspond to the outputs defined in their [Algo](./algo.md).
Each compute task output has a separate set of [Permissions](../permissions.md).
The permissions must be set for each output at task creation,
with the exception of outputs of kind "performance" which are always public and for which setting specific permissions is not allowed.

## Compatibility

> Tasks receives explicit inputs, they can be either:

Reviewer: ```suggestion
Tasks receive explicit inputs, they can be either:
```

- a direct reference to an existing asset
- or a reference to another task's output

On task creation, inputs should match the expectations of the algorithm used:

- mandatory inputs should be defined
- asset kind should match

> - permissions should allow processing

Reviewer 2: This item feels different from the others. Maybe move it to a standalone sentence?

> For legacy reasons, a task cannot receive more than one datamanager

Reviewer 2: I don't think it's legacy, it's more like the current (intentional) design. What do you think?

```suggestion
A task cannot have than one datamanager as an input.
```
